### PR TITLE
Special (Pickable) quests + Wallet Quests

### PR DESCRIPTION
**Pull request by kevinunger**
FE work for #1458 and issue #1447 
Still wip

### CODE REVIEW

**Comment by kevinunger** [comment]
image
image
image

**Comment by poshdan** [comment]
Committed some fixes to the contract.
Added migration 178 to set up some test pickable & wallet quests. (external quests use weapons (submit) and junk (hold) )
There will be merge conflicts with main branch, these pertain to Frontend, i could not resolve them myself.

I have to pass this along to someone working on frontend.
Key notes:

-ditch the indexInTier param for wallet quests
-pickable quests now also specify the questID instead of index
-force 0 rep and disable stamina/raid inputs and experience output for wallet quests
-add option on skip or completion to then pick a questID from the pickable tier (if that tier is not empty)
-in case it's not done yet, implement character reward display
-chances are, external hold is not implemented in frontend, just like how it wasn't implemented in solidity.

It may be needed to visualize which IDs of an NFT series have already been "shown" (aka proven held).
Basically, individual NFTs may only be used to complete a hold quest once, but they are not given away (unlike with the regular external (submit) quests)

Use PartnerVault.haveNftsBeenShown(address tokenAddress, uint256[] memory tokenIds)
to check which IDs have been shown before

**Comment by ed3ath** [comment]
fix blockers: 
- to complete / request wallet quests dan used a indexInTier param, which is now removed. Need to check how it works now to select the wallet quests

requesting review for contract change @poshdan 
https://github.com/CryptoBlades/cryptoblades/pull/1480/commits/a101246ec0f0126eb852f69f79ecfc9d012ba3e1#diff-1d78d4f27bdd58f46b7a16926959e189b9290eb0be2840c750be20e5b7776a14

**Review by poshdan — APPROVED** [review]
